rfi_file_monitor/operations/scicataloguer.py

```
@@ -411,34 +424,21 @@ def _check_required_fields(params):
         if not params.owner_group:
             raise RequiredInfoNotFound("Owner group required")
 
-    def checkbox_toggled(self, checkbox):
-        # Set class attribute for derived/raw dataset
-        if checkbox.get_active() == True:
-            self.params.derived_dataset = True
-            self._input_datasets_entry.set_sensitive(True)
-            self._used_software_entry.set_sensitive(True)
-            return True
-        elif checkbox.get_active() == False:
-            self.params.derived_dataset = False
-            self._input_datasets_entry.set_sensitive(False)
-            self._used_software_entry.set_sensitive(False)
-            return False
-
     def preflight_check(self):
+        self._preflight_check(self.params)
+
+    def _preflight_check(self, params):
         try:
             ScicatClient(
-                base_url=self.params.hostname,
-                username=self.params.username,
-                password=self.params.password,
+                base_url=params.hostname,
+                username=params.username,
+                password=params.password,
             )
         except Exception as e:
             logger.error(f"Could not login to scicat: {e}")
+            return str(e)
```

raise this error rather than return a string to allow appwindow to display